### PR TITLE
Feat/mpp 286 retrieve data on 10 past purchases

### PR DESCRIPTION
**Pull request by joyet-simon**
## Reason for change

[Linear task](https://linear.app/almapay/issue/MPP-286/sfcc-retrieve-data-on-10-past-purchases)

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

**Comment by ClaireAlma** [comment]
You forgot the how to test part of the PR description

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 1]
```diff
@@ -159,6 +159,95 @@ function haveExcludedCategory(productIds) {
...
+            fullPageUrl = getFullPageUrl(product.getPageURL(), product.getMasterProduct().getID(), locale);
+            productsCategories = product.getMasterProduct().getAllCategories().toArray();
+        }
+
+        productsCategories.forEach(function (category) {
+            if (!categories.includes(category.getID())) {
+                categories.push(category.getID());
+            }
+        });
+        var item = {
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:204]: create a function buildItem something like that, try to reduce the complexity of your functions and create small dedicated functions

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 2]
```diff
@@ -159,6 +159,95 @@ function haveExcludedCategory(productIds) {
...
+function getWebsiteCustomerDetails(customer, locale) {
+    var forOf = require('*/cartridge/scripts/helpers/almaUtilsHelper').forOf;
+    var isGuest = customer.isAnonymous();
+
+    var previousOrders = [];
+
+    if (!isGuest) {
+        var orders = customer.getOrderHistory().getOrders().asList(0, 10);
+        forOf(orders, function (order) {
+            var previourOrder = {
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:235]: add function build/getPreviousOrder

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

**Review by Francois-Gomis — COMMENTED** [review]
Same

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:204]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 3]
```diff
@@ -159,6 +159,95 @@ function haveExcludedCategory(productIds) {
...
+function getOrdersItemsForWebsiteCustomerDetails(order, locale) {
+    var forOf = require('*/cartridge/scripts/helpers/almaUtilsHelper').forOf;
+    var items = [];
+    forOf(order.getAllProductLineItems(), function (productLine) {
+        var product = productLine.getProduct();
+        var categories = [];
+        var fullPageUrl = '';
+        var productsCategories = '';
+
+        if (product.isMaster()) {
```
[the remark this answers is not in the document]
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:191]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 1 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:204]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 2 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:235]: Done

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,2 @@
+* simom.joyet@getalma.eu
```
Comment (Francois-Gomis) [inline .github/CODEOWNERS:1]: simom ?

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 5]
```diff
@@ -159,6 +159,123 @@ function haveExcludedCategory(productIds) {
...
+    if (product.isMaster()) {
+        productId = product.getID();
+    } else {
+        productId = product.getMasterProduct().getID();
+    }
+    return 'https://' + Site.getCurrent().getHttpsHostName() + '/s/' + Site.getCurrent().getName() + '/' + product.getPageURL() + '/' + productId + '.html?lang=' + locale;
+}
+
+/**
+ * Get catégories for a product
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:179]: catégories in french

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 6]
```diff
@@ -159,6 +159,123 @@ function haveExcludedCategory(productIds) {
...
+    return items;
+}
+
+/**
+ * Get previous order
+ * @param {Object} order order
+ * @param {string} locale locale
+ * @returns {Object} previous order
+ */
+function getPreviousOrder(order, locale) {
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:245]: it's a format function?

**Review by Francois-Gomis — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 4 above)
Comment (joyet-simon) [reply .github/CODEOWNERS]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 5 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 6 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js]: Done

**Review by Francois-Gomis — APPROVED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js [hunk 7]
```diff
@@ -159,6 +159,123 @@ function haveExcludedCategory(productIds) {
...
     return haveExcludedCategoryReturn;
 }
+/**
+ * Get the full url for a page
+ * @param {Object} product product
+ * @param {string} locale locale
+ * @returns {string} url
+ */
+function getFullPageUrl(product, locale) {
+    var productId = '';
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js:169]: add a getProductId function

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js (hunk 7 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js]: Done

**Comment by ClaireAlma** [comment]
Please describe better the pr, how to test etc...

File: cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+/**
+ * Get product id
+ * @param {Object} product product
+ * @returns {string} id
+ */
+function getProductId(product) {
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js:8]: /**
 * Get product id
 * @param {Object} product product
 * @returns {string} id
 */
function getProductId(product) {
       if (product.isMaster()) {
      return  product.getID();
    } 
    return  product.getMasterProduct().getID();
    
}

File: cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    var productId = '';
+
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js:10]: pareil que la fonction d'avant

File: cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+/**
+ * Get categories for a product
+ * @param {Object} product product
+ * @returns {array} categories
+ */
+function getProductCategories(product) {
+    var productsCategories = [];
+
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js:27]: pareil

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

File: cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js (hunk 9 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js]: Done

File: cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js (hunk 10 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/scripts/helpers/almaProductHelper.js]: Done

**Review by ClaireAlma — APPROVED** [review]

**Pull request merged**